account_reconciliation_widget/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1603,7 +1603,10 @@ odoo.define("account.ReconciliationModel", function (require) {
                     );
                     if (matching && matching.length === 2) {
                         matching = matching[1].replace(
-                            new RegExp("\\D" + reconcileModel.decimal_separator, "g"),
+                            new RegExp(
+                                "[^-d" + reconcileModel.decimal_separator + "]",
```

Comment:
Missing backslash.
Also in the v14 patch. https://github.com/OCA/account-reconcile/pull/420/files#r920156214

Reply:
Fixed for v15

Reply:
finally,  we need \\d @jordens